pkg/stanza/fileconsumer/file.go

```diff
@@ -294,6 +294,11 @@ func (m *Manager) findFingerprintMatch(fp *Fingerprint) (*Reader, bool) {
 	for i := len(m.knownFiles) - 1; i >= 0; i-- {
 		oldReader := m.knownFiles[i]
 		if fp.StartsWith(oldReader.Fingerprint) {
+			// Remove the old reader from the list of known files. We will
+			// add it back in saveCurrent if it is still alive.
+			defer func() {
```

Comment:
Why is deferring needed here, given that the next step is return?

Reply:
You're right it's not. I was concerned about mutating a slice as we iterate over it, but it is not a problem here.